src/surfaceintersection.jl

```diff
@@ -0,0 +1,111 @@
+surfacecrossing{T<:Real}(g::Taylor1{T}, g_old::Taylor1{T}, order::Int) = g[order+1]*g_old[order+1] < zero(T)
+surfacecrossing{T<:Real}(g::Taylor1{Taylor1{T}}, g_old::Taylor1{Taylor1{T}}, order::Int) = g[order+1][1]*g_old[order+1][1] < zero(T)
+surfacecrossing{T<:Real}(g::Taylor1{TaylorN{T}}, g_old::Taylor1{TaylorN{T}}, order::Int) = g[order+1][1][1]*g_old[order+1][1][1] < zero(T)
+
+function deriv{T<:Number}(n::Int, a::Taylor1{T})
```

Comment:
Why do you prefer this method over `derivative(n,a)`?

Reply:
After discussing this offline with @lbenet, now I think this should co-exist with `derivative(n,a)` in TaylorSeries. Let me explain myself:

The main difference is that, as it stands now, `derivative` returns the *numerical value* of the `n`-th derivative of `a`, at the original point of expansion. But my proposal is that we should have a method (such as what I have called `deriv`) which should return the `n`-th derivative of `a` as a Taylor *polynomial*. And, only if evaluated at `zero`, it should return the *numerical value* of the `n`-th derivative of `a`, at the original point of expansion. But that's material for another PR for `TaylorSeries.jl`, which I'll submit asap!